### PR TITLE
Make sure AwsClient is useful

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: src/Core/AbstractApi.php [hunk 1]
```diff
@@ -167,7 +167,7 @@ private function stringifyBody($body): string
         return (string) $body;
     }
 
-    private function getSigner()
+    private function getSigner(?string $scope = null)
     {
         if (null === $this->signer) {
```
Comment (jderusse) [inline src/Core/AbstractApi.php]: the signer should vary with method's argument.

using the same AwsClient to perform 2 requests in 2 services will failed.

**Comment by Nyholm** [comment]
Hm.. maybe `AwsClient` should not extend `AbstractClient`. That will remove the "feature" of being allowed to make requests with it. 

The `AwsClient` will then only work as a service storage.

File: src/Core/AwsClient.php [hunk 2]
```diff
@@ -4,33 +4,93 @@
...
+        if ($configuration->isDefault(Configuration::OPTION_ENDPOINT)) {
+            $this->credentialProvider = $credentialProvider ?? new CacheProvider(new ChainProvider([
+                new ConfigurationProvider(),
+                new IniFileProvider($this->logger),
+                new InstanceProvider($this->httpClient, $this->logger),
+            ]));
+        } else {
+            $this->credentialProvider = $credentialProvider ?? new CacheProvider(new ChainProvider([
+                new ConfigurationProvider(),
+                new IniFileProvider($this->logger),
```
Comment (Nyholm) [inline src/Core/AwsClient.php]: Could we refactor this constructor code somehow?

Comment (jderusse) [reply src/Core/AwsClient.php]: fixed by #190

Comment (jderusse) [reply src/Core/AwsClient.php]: remove the `else` part and just keep the `if`

Comment (Nyholm) [reply src/Core/AwsClient.php]: Like this?

Comment (jderusse) [reply src/Core/AwsClient.php]: :+1:

**Pull request merged**